### PR TITLE
fix setup issues for the release/v1.16 branch

### PR DESCRIPTION
**Pull request by vbaddi**
- Fix the setup issues, to be inline with the deployment.
- Over-ride the models/qpc directory using the env variable. -- `export QEFF_HOME`

### CODE REVIEW

**Comment by vbaddi** [comment]
cc @quic-rishinr @carlstreeter-quic @quic-jhugo

**Review by carlstreeter-quic — APPROVED** [review]
tested in qaic-docker

File: QEfficient/utils/constants.py [hunk 1]
```diff
@@ -10,7 +10,41 @@
...
+
+    Sets QEFF_MODELS_DIR environment variable if not already set.
+
+    Returns:
+        str: Path to the QEFF models directory.
+    """
+    qeff_cache_home = os.environ.get("QEFF_HOME")
+    # Check if XDG_CACHE_HOME is set
+    xdg_cache_home = os.environ.get("XDG_CACHE_HOME")
+    if xdg_cache_home:
```
Comment (irajagop) [inline QEfficient/utils/constants.py:33]: It should be other way around, first check for QEFF_HOME, only if not present you have to check for XDG_CACHE_HOME.

File: QEfficient/utils/constants.py [hunk 2]
```diff
@@ -10,7 +10,41 @@
...
+        qeff_models_dir = os.path.join(xdg_cache_home, "qeff_models")
+    # Check if QEFF_MODELS_DIR is set
+    elif qeff_cache_home:
+        qeff_models_dir = os.path.join(qeff_cache_home, "qeff_models")
```
Comment (irajagop) [inline QEfficient/utils/constants.py:37]: We should directly assign `qeff_models_dir = qeff_cache_home`

File: QEfficient/utils/constants.py [hunk 3]
```diff
@@ -10,7 +10,41 @@
...
+    else:
+        # Use ~/.cache/qeff_models as the default
+        qeff_models_dir = os.path.join(os.path.expanduser("~"), ".cache", "qeff_models")
+
+    # Set QEFF_MODELS_DIR environment variable
+    os.environ["QEFF_HOME"] = qeff_models_dir
```
Comment (irajagop) [inline QEfficient/utils/constants.py:43]: We shouldn't set the environment variable. This is available only in python env, and will be reset when python exits.

File: QEfficient/utils/constants.py [hunk 4]
```diff
@@ -10,7 +10,41 @@
 UTILS_DIR = os.path.dirname(os.path.abspath(__file__))
 QEFF_DIR = os.path.dirname(UTILS_DIR)
 ROOT_DIR = os.path.dirname(QEFF_DIR)
```
Comment (irajagop) [inline QEfficient/utils/constants.py:12]: We should remove these too, as these again create files in the installation path.

File: QEfficient/utils/constants.py (hunk 1 above)
Comment (vbaddi) [reply QEfficient/utils/constants.py]: Done

File: QEfficient/utils/constants.py (hunk 3 above)
Comment (vbaddi) [reply QEfficient/utils/constants.py]: Agree, Removed

File: QEfficient/utils/constants.py (hunk 4 above)
Comment (vbaddi) [reply QEfficient/utils/constants.py:12]: There is some dependency on other scripts for ROOT_DIR, would like to have this until fully tested.

File: QEfficient/utils/constants.py (hunk 2 above)
Comment (vbaddi) [reply QEfficient/utils/constants.py:34]: The default path for all file save should be `qeff_models`, along with QEFF_HOME/XDG_CACHE_HOME/~.cache.

File: QEfficient/utils/constants.py (hunk 4 above)
Comment (vbaddi) [reply QEfficient/utils/constants.py:12]: as mentioned in #68 same goes here too, this requires changes in many places making the review difficult, will take this up once this is closed. thanks, resolving the chat.

**Review by carlstreeter-quic — APPROVED** [review]
Tested QEfficient.cloud.infer in docker container with locked down venv, and that works now, too.

**Comment by vbaddi** [comment]
Priority for saving the export onnx/qpc paths:

- Use QEFF_HOME if set in environment.
- Use $XDG_CACHE_HOME/qeff_models if XDG_CACHE_HOME is set.
- Default to ~/.cache/qeff_models.

**Pull request merged**